test/jaxp/javax/xml/jaxp/unittest/validation/ValidationTest.java

```diff
@@ -139,6 +142,52 @@ public void startElement(String uri, String localName, String qName, Attributes
 
     }
 
+    /**
+     * Verifies the bug fix for 8349516. The fix adds a guard against empty text
+     * since calling StreamReader.getTextCharacters with textLength=0 will result
+     * in IndexOutOfBoundsException.
+     *
+     * @throws Exception if the test fails, in which case the parser throws
+     * IndexOutOfBoundsException.
```

Comment:
Do we need this `throws Exception` clause (and the method declaration below)? Since AIOOBE is a `RuntimeException`, this `throws` clause does not seem to be necessary. Simple description of the failure case may suffice.

Reply:
Thanks Naoto. Improved the javadoc a bit.